### PR TITLE
Allow client auth name/key to be read directly from a file

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Blocked while testing with a real client.

File: src/swarm/neo/authentication/Credentials.d [hunk 1]
```diff
@@ -11,98 +11,52 @@
...
-        Ensures this instance complies to the limits.
+    Returns:
+        name.length minus the index of the first non-graph character found
+        or 0 if all characters in name are graph characters.
 
-    ***************************************************************************/
+*******************************************************************************/
 
-    invariant ( )
+public static size_t validateNameCharacters ( cstring name )
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/authentication/Credentials.d]: `static` is now extra

File: src/swarm/neo/authentication/CredentialsFile.d [hunk 2]
```diff
@@ -38,275 +38,219 @@
...
+public HmacDef.Key[istring] parse ( cstring filepath )
+{
+    scope file = new File(filepath, File.ReadExisting);
+    scope ( exit ) file.close;
 
-    ***************************************************************************/
+    size_t file_length = file.length;
 
-    public this ( istring filepath )
+    if (file_length > CredDef.LengthLimit.File)
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/authentication/CredentialsFile.d:91]: I feel we can work around this limit in the future - it doesn't strike me like a something that's a necessity. Not related to this PR.

**Comment by nemanja-boric-sociomantic** [comment]
Looks good!

**Comment by gavin-norman-sociomantic** [comment]
Tested. Works as expected. Unblocked.

**Comment by gavin-norman-sociomantic** [comment]
Updated:
* Removed `static` from free function.

File: src/swarm/neo/authentication/CredentialsFile.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/authentication/CredentialsFile.d:91]: Yeah, I was also surprised to find this limit. It's a simplistic implementation that we can extend at some point, if needed.

**Comment by nemanja-boric-sociomantic** [comment]
LGTM

**Pull request merged**